No files found during mirroring IT

https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/135853

```
======================================================================
FAIL: test_indexing (integration_test.IndexingIntegrationTest.test_indexing) [mirror_sources_and_files]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 400, in subTest
    yield
  File "/builds/ucsc/azul/test/integration_test.py", line 1820, in _test_mirroring
    repository_file, source, file_response = self._get_one_mirrorable_file(catalog)
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/builds/ucsc/azul/test/integration_test.py", line 771, in _get_one_mirrorable_file
    outer_file, inner_file = self._get_one_file(catalog,
                             ~~~~~~~~~~~~~~~~~~^^^^^^^^^
                                                max_size=config.catalogs[catalog].mirror_limit)
                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/test/integration_test.py", line 815, in _get_one_file
    self.fail('No files found')
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^
AssertionError: No files found
```

```
2026-02-28T23:39:31.335214Z 01E 2026-02-28 23:39:31,334    INFO MainThread test.integration_test: Beginning sub-test [mirror_sources_and_files] {}
2026-02-28T23:39:31.337294Z 01E 2026-02-28 23:39:31,336    INFO MainThread azul.oauth2: Making GET request to 'https://service.sandbox.dev.singlecell.gi.ucsc.edu/index/files?catalog=dcp3-it&filters=%7B%22fileFormat%22%3A+%7B%22is%22%3A+%5B%22fastq%22%2C+%22fastq.gz%22%5D%7D%2C+%22fileSize%22%3A+%7B%22within%22%3A+%5B%5B1%2C+1610612736%5D%5D%7D%7D&size=1&order=asc&sort=fileSize'
2026-02-28T23:39:31.337358Z 01E 2026-02-28 23:39:31,337    INFO MainThread azul.oauth2: … without a request body
2026-02-28T23:39:31.338458Z 01E 2026-02-28 23:39:31,338    INFO MainThread azul.oauth2: … with request headers [('x-goog-api-client', 'cred-type/sa'), ('authorization', 'REDACTED')]
2026-02-28T23:39:31.553925Z 01E 2026-02-28 23:39:31,553    INFO MainThread azul.oauth2: Got 200 response after 0.216s from GET to https://service.sandbox.dev.singlecell.gi.ucsc.edu/index/files?catalog=dcp3-it&filters=%7B%22fileFormat%22%3A+%7B%22is%22%3A+%5B%22fastq%22%2C+%22fastq.gz%22%5D%7D%2C+%22fileSize%22%3A+%7B%22within%22%3A+%5B%5B1%2C+1610612736%5D%5D%7D%7D&size=1&order=asc&sort=fileSize
2026-02-28T23:39:31.554343Z 01E 2026-02-28 23:39:31,553    INFO MainThread azul.oauth2: … with response headers HTTPHeaderDict({'Content-Type': 'application/json', 'Content-Length': '2323', 'Connection': 'keep-alive', 'Date': 'Sat, 28 Feb 2026 23:39:31 GMT', 'X-Amzn-Trace-Id': 'Root=1-69a37cb3-41e5dffb1e2d04cd667dccab;Parent=32a3f1d2e25c0bca;Sampled=0;Lineage=1:bf371b33:0', 'x-amzn-RequestId': 'f96f3211-24da-4e36-8413-47a6f51523cc', 'Referrer-Policy': 'strict-origin-when-cross-origin', 'X-XSS-Protection': '1; mode=block', 'Access-Control-Allow-Origin': '*', 'Strict-Transport-Security': 'max-age=63072000; includeSubDomains; preload', 'Access-Control-Allow-Headers': 'Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key', 'X-Frame-Options': 'DENY', 'Content-Security-Policy': "default-src 'self';img-src 'self' data:;script-src 'self';style-src 'self';frame-ancestors 'none';form-action 'self'", 'x-amz-apigw-id': 'ZhBsFH9ToAMECZw=', 'Cache-Control': 'no-store', 'X-Content-Type-Options': 'nosniff', 'X-Cache': 'Miss from cloudfront', 'Via': '1.1 6e3cb302b7b47cb215ef7af27e526b74.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'IAD61-P13', 'X-Amz-Cf-Id': '3QuYtdhSTCmrPj7iHKAZJUlCs4P9NhWibe80FSRvGv3OjAy1qSX25A=='})
2026-02-28T23:39:31.554352Z 01E 2026-02-28 23:39:31,553    INFO MainThread azul.oauth2: … with a response body of length 2323 starting in b'{"pagination":{"count":0,"total":0,"size":1,"next":null,"previous":null,"pages":0,"sort":"fileSize","order":"asc"},"termFacets":{"organ":{"terms":[],"total":0,"type":"terms"},"sampleEntityType":{"terms":[],"total":0,"type":"terms"},"dataUseRestriction":{"terms":[],"total":0,"type":"terms"},"project":{"terms":[],"total":0,"type":"terms"},"sampleDisease":{"terms":[],"total":0,"type":"terms"},"nucleicAcidSource":{"terms":[],"total":0,"type":"terms"},"assayType":{"terms":[],"total":0,"type":"terms"},"instrumentManufacturerModel":{"terms":[],"total":0,"type":"terms"},"institution":{"terms":[],"total":0,"type":"terms"},"donorDisease":{"terms":[],"total":0,"type":"terms"},"organismAge":{"terms":[],"total":0,"type":"terms"},"contentDescription":{"terms":[],"total":0,"type":"terms"},"pairedEnd":{"terms":[],"total":0,"type":"terms"},"developmentStage":{"terms":[],"total":0,"type":"terms"},"preservationMethod":{"terms":[],"total":0,"type":"terms"},"genusSpecies":{"terms":[],"total":0,"type":"terms"},"projectTitle":{"...'
2026-02-28T23:39:31.555603Z 01E 2026-02-28 23:39:31,555    INFO MainThread azul.oauth2: Making GET request to 'https://service.sandbox.dev.singlecell.gi.ucsc.edu/index/files?catalog=dcp3-it&filters=%7B%22fileSize%22%3A+%7B%22within%22%3A+%5B%5B1%2C+1610612736%5D%5D%7D%7D&size=1&order=asc&sort=fileSize'
2026-02-28T23:39:31.555619Z 01E 2026-02-28 23:39:31,555    INFO MainThread azul.oauth2: … without a request body
2026-02-28T23:39:31.556235Z 01E 2026-02-28 23:39:31,555    INFO MainThread azul.oauth2: … with request headers [('x-goog-api-client', 'cred-type/sa'), ('authorization', 'REDACTED')]
2026-02-28T23:39:31.701112Z 01E 2026-02-28 23:39:31,693    INFO MainThread azul.oauth2: Got 200 response after 0.138s from GET to https://service.sandbox.dev.singlecell.gi.ucsc.edu/index/files?catalog=dcp3-it&filters=%7B%22fileSize%22%3A+%7B%22within%22%3A+%5B%5B1%2C+1610612736%5D%5D%7D%7D&size=1&order=asc&sort=fileSize
2026-02-28T23:39:31.701128Z 01E 2026-02-28 23:39:31,694    INFO MainThread azul.oauth2: … with response headers HTTPHeaderDict({'Content-Type': 'application/json', 'Content-Length': '2349', 'Connection': 'keep-alive', 'Date': 'Sat, 28 Feb 2026 23:39:31 GMT', 'X-Amzn-Trace-Id': 'Root=1-69a37cb3-184dec4562e998385122f936;Parent=6396ac09afc91564;Sampled=0;Lineage=1:bf371b33:0', 'x-amzn-RequestId': '4f4a3604-40d9-411c-8e3d-9f31df296732', 'Referrer-Policy': 'strict-origin-when-cross-origin', 'X-XSS-Protection': '1; mode=block', 'Access-Control-Allow-Origin': '*', 'Strict-Transport-Security': 'max-age=63072000; includeSubDomains; preload', 'Access-Control-Allow-Headers': 'Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key', 'X-Frame-Options': 'DENY', 'Content-Security-Policy': "default-src 'self';img-src 'self' data:;script-src 'self';style-src 'self';frame-ancestors 'none';form-action 'self'", 'x-amz-apigw-id': 'ZhBsHH5MoAMEJgA=', 'Cache-Control': 'no-store', 'X-Content-Type-Options': 'nosniff', 'X-Cache': 'Miss from cloudfront', 'Via': '1.1 6e3cb302b7b47cb215ef7af27e526b74.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'IAD61-P13', 'X-Amz-Cf-Id': 'QeJbhLp5D94x0Ld9K5E0ts0E-BJd7FWOnjlT5PMy7Wp_vSrWzN4ahA=='})
2026-02-28T23:39:31.701154Z 01E 2026-02-28 23:39:31,694    INFO MainThread azul.oauth2: … with a response body of length 2349 starting in b'{"pagination":{"count":0,"total":0,"size":1,"next":null,"previous":null,"pages":0,"sort":"fileSize","order":"asc"},"termFacets":{"organ":{"terms":[],"total":0,"type":"terms"},"sampleEntityType":{"terms":[],"total":0,"type":"terms"},"dataUseRestriction":{"terms":[],"total":0,"type":"terms"},"project":{"terms":[],"total":0,"type":"terms"},"sampleDisease":{"terms":[],"total":0,"type":"terms"},"nucleicAcidSource":{"terms":[],"total":0,"type":"terms"},"assayType":{"terms":[],"total":0,"type":"terms"},"instrumentManufacturerModel":{"terms":[],"total":0,"type":"terms"},"institution":{"terms":[],"total":0,"type":"terms"},"donorDisease":{"terms":[],"total":0,"type":"terms"},"organismAge":{"terms":[],"total":0,"type":"terms"},"contentDescription":{"terms":[],"total":0,"type":"terms"},"pairedEnd":{"terms":[],"total":0,"type":"terms"},"developmentStage":{"terms":[],"total":0,"type":"terms"},"preservationMethod":{"terms":[],"total":0,"type":"terms"},"genusSpecies":{"terms":[],"total":0,"type":"terms"},"projectTitle":{"...'
2026-02-28T23:39:31.701179Z 01E 2026-02-28 23:39:31,694    INFO MainThread test.integration_test: Failed sub-test [mirror_sources_and_files] {}
2026-02-28T23:39:31.701181Z 01E   test_indexing (integration_test.IndexingIntegrationTest.test_indexing) [mirror_sources_and_files] ... FAIL
```

--- a/UPGRADING.rst
+++ b/UPGRADING.rst
@@ -20,6 +20,20 @@ reverted. This is all fairly informal and loosely defined. Hopefully we won't
 have too many entries in this file.
 
 
+#7815 No files found during mirroring IT
+========================================
+
+Flag the following sources as ``no_mirror`` in your personal deployments
+colocated with ``dev``:
+
+ - ``datarepo-dev-3041c2cf:hca_dev_2af52a1365cb4973b51339be38f2df3f__20210830_20210903``
+ - ``datarepo-dev-40283c27:hca_dev_3a69470330844ece9abed935fd5f6748__20210901_20210903``
+ - ``datarepo-dev-788c3b52:hca_dev_955dfc2ca8c64d04aa4d907610545d11__20210831_20210903``
+
+As always, use the sandbox deployment's ``environment.py`` as a model when
+upgrading personal deployments.
+
+
 #6675 Docker build with Bookworm base image fails on GL
 =======================================================
 

--- a/deployments/dev/environment.py
+++ b/deployments/dev/environment.py
@@ -61,6 +61,7 @@ def union(previous_catalog: dict[DatasetName, SourceItem | None],
 
 
 dcp2_sources = union({}, 105, delta([
+    # @formatter:off
     source('bigquery', 'datarepo-dev-a9252919', 'hca_dev_005d611a14d54fbf846e571a1f874f70__20210827_20210903'),
     source('bigquery', 'datarepo-dev-c148d39c', 'hca_dev_027c51c60719469fa7f5640fe57cbece__20210827_20210902'),
     source('bigquery', 'datarepo-dev-e2ab8487', 'hca_dev_03c6fce7789e4e78a27a664d562bb738__20210902_20210907'),
@@ -84,12 +85,12 @@ dcp2_sources = union({}, 105, delta([
     source('bigquery', 'datarepo-dev-1c2c69d9', 'hca_dev_24c654a5caa5440a8f02582921f2db4a__20210830_20210903'),
     source('bigquery', 'datarepo-dev-38f08cd8', 'hca_dev_2a64db431b554639aabb8dba0145689d__20210830_20210903'),
     source('bigquery', 'datarepo-dev-4cf05ce2', 'hca_dev_2a72a4e566b2405abb7c1e463e8febb0__20210901_20210903'),
-    source('bigquery', 'datarepo-dev-3041c2cf', 'hca_dev_2af52a1365cb4973b51339be38f2df3f__20210830_20210903'),
+    source('bigquery', 'datarepo-dev-3041c2cf', 'hca_dev_2af52a1365cb4973b51339be38f2df3f__20210830_20210903', no_mirror),  # noqa E501
     source('bigquery', 'datarepo-dev-311340f6', 'hca_dev_2d8460958a334f3c97d4585bafac13b4__20210902_20210907'),
     source('bigquery', 'datarepo-dev-766bfb76', 'hca_dev_2ef3655a973d4d699b4121fa4041eed7__20210827_20210903'),
     source('bigquery', 'datarepo-dev-1720b3c5', 'hca_dev_379ed69ebe0548bcaf5ea7fc589709bf__20210827_20210902'),
     source('bigquery', 'datarepo-dev-ac6efd3f', 'hca_dev_38449aea70b540db84b31e08f32efe34__20210830_20210903'),
-    source('bigquery', 'datarepo-dev-40283c27', 'hca_dev_3a69470330844ece9abed935fd5f6748__20210901_20210903'),
+    source('bigquery', 'datarepo-dev-40283c27', 'hca_dev_3a69470330844ece9abed935fd5f6748__20210901_20210903', no_mirror),  # noqa E501
     source('bigquery', 'datarepo-dev-b08233fa', 'hca_dev_3cfcdff5dee14a7ba591c09c6e850b11__20210827_20210903'),
     source('bigquery', 'datarepo-dev-bdc9f342', 'hca_dev_3e329187a9c448ec90e3cc45f7c2311c__20210901_20210903'),
     source('bigquery', 'datarepo-dev-ec07c8d8', 'hca_dev_4037007b0eff4e6db7bd8dd8eec80143__20210831_20210903'),
@@ -123,7 +124,7 @@ dcp2_sources = union({}, 105, delta([
     source('bigquery', 'datarepo-dev-0c5c20b5', 'hca_dev_8c3c290ddfff4553886854ce45f4ba7f__20210902_20210907'),
     source('bigquery', 'datarepo-dev-29509483', 'hca_dev_90bd693340c048d48d76778c103bf545__20210827_20211110'),
     source('bigquery', 'datarepo-dev-59d37b9a', 'hca_dev_946c5add47d1402a97bba5af97e8bce7__20210831_20210903'),
-    source('bigquery', 'datarepo-dev-788c3b52', 'hca_dev_955dfc2ca8c64d04aa4d907610545d11__20210831_20210903'),
+    source('bigquery', 'datarepo-dev-788c3b52', 'hca_dev_955dfc2ca8c64d04aa4d907610545d11__20210831_20210903', no_mirror),  # noqa E501
     source('bigquery', 'datarepo-dev-4b88b45b', 'hca_dev_962bd805eb894c54bad2008e497d1307__20210830_20210903'),
     source('bigquery', 'datarepo-dev-02c59b72', 'hca_dev_99101928d9b14aafb759e97958ac7403__20210830_20210903'),
     source('bigquery', 'datarepo-dev-a6312a94', 'hca_dev_992aad5e7fab46d9a47ddf715e8cfd24__20210830_20210903'),
@@ -166,6 +167,7 @@ dcp2_sources = union({}, 105, delta([
     source('bigquery', 'datarepo-dev-67240cf2', 'hca_dev_f86f1ab41fbb4510ae353ffd752d4dfc__20210901_20210903'),
     source('bigquery', 'datarepo-dev-e8e0a59a', 'hca_dev_f8aa201c4ff145a4890e840d63459ca2__20210901_20210903'),
     source('bigquery', 'datarepo-dev-96d8e08c', 'hca_dev_faeedcb0e0464be7b1ad80a3eeabb066__20210831_20210903'),
+    # @formatter:on
 ]))
 
 dcp3_sources = union(dcp2_sources, 111, delta([

--- a/deployments/sandbox/environment.py
+++ b/deployments/sandbox/environment.py
@@ -63,6 +63,7 @@ def union(previous_catalog: dict[DatasetName, SourceItem | None],
 
 
 dcp2_sources = union({}, 105, delta([
+    # @formatter:off
     source('bigquery', 'datarepo-dev-a9252919', 'hca_dev_005d611a14d54fbf846e571a1f874f70__20210827_20210903'),
     source('bigquery', 'datarepo-dev-c148d39c', 'hca_dev_027c51c60719469fa7f5640fe57cbece__20210827_20210902'),
     source('bigquery', 'datarepo-dev-e2ab8487', 'hca_dev_03c6fce7789e4e78a27a664d562bb738__20210902_20210907'),
@@ -86,12 +87,12 @@ dcp2_sources = union({}, 105, delta([
     source('bigquery', 'datarepo-dev-1c2c69d9', 'hca_dev_24c654a5caa5440a8f02582921f2db4a__20210830_20210903'),
     source('bigquery', 'datarepo-dev-38f08cd8', 'hca_dev_2a64db431b554639aabb8dba0145689d__20210830_20210903'),
     source('bigquery', 'datarepo-dev-4cf05ce2', 'hca_dev_2a72a4e566b2405abb7c1e463e8febb0__20210901_20210903'),
-    source('bigquery', 'datarepo-dev-3041c2cf', 'hca_dev_2af52a1365cb4973b51339be38f2df3f__20210830_20210903'),
+    source('bigquery', 'datarepo-dev-3041c2cf', 'hca_dev_2af52a1365cb4973b51339be38f2df3f__20210830_20210903', no_mirror),  # noqa E501
     source('bigquery', 'datarepo-dev-311340f6', 'hca_dev_2d8460958a334f3c97d4585bafac13b4__20210902_20210907'),
     source('bigquery', 'datarepo-dev-766bfb76', 'hca_dev_2ef3655a973d4d699b4121fa4041eed7__20210827_20210903'),
     source('bigquery', 'datarepo-dev-1720b3c5', 'hca_dev_379ed69ebe0548bcaf5ea7fc589709bf__20210827_20210902'),
     source('bigquery', 'datarepo-dev-ac6efd3f', 'hca_dev_38449aea70b540db84b31e08f32efe34__20210830_20210903'),
-    source('bigquery', 'datarepo-dev-40283c27', 'hca_dev_3a69470330844ece9abed935fd5f6748__20210901_20210903'),
+    source('bigquery', 'datarepo-dev-40283c27', 'hca_dev_3a69470330844ece9abed935fd5f6748__20210901_20210903', no_mirror),  # noqa E501
     source('bigquery', 'datarepo-dev-b08233fa', 'hca_dev_3cfcdff5dee14a7ba591c09c6e850b11__20210827_20210903'),
     source('bigquery', 'datarepo-dev-bdc9f342', 'hca_dev_3e329187a9c448ec90e3cc45f7c2311c__20210901_20210903'),
     source('bigquery', 'datarepo-dev-ec07c8d8', 'hca_dev_4037007b0eff4e6db7bd8dd8eec80143__20210831_20210903'),
@@ -125,7 +126,7 @@ dcp2_sources = union({}, 105, delta([
     source('bigquery', 'datarepo-dev-0c5c20b5', 'hca_dev_8c3c290ddfff4553886854ce45f4ba7f__20210902_20210907'),
     source('bigquery', 'datarepo-dev-29509483', 'hca_dev_90bd693340c048d48d76778c103bf545__20210827_20211110'),
     source('bigquery', 'datarepo-dev-59d37b9a', 'hca_dev_946c5add47d1402a97bba5af97e8bce7__20210831_20210903'),
-    source('bigquery', 'datarepo-dev-788c3b52', 'hca_dev_955dfc2ca8c64d04aa4d907610545d11__20210831_20210903'),
+    source('bigquery', 'datarepo-dev-788c3b52', 'hca_dev_955dfc2ca8c64d04aa4d907610545d11__20210831_20210903', no_mirror),  # noqa E501
     source('bigquery', 'datarepo-dev-4b88b45b', 'hca_dev_962bd805eb894c54bad2008e497d1307__20210830_20210903'),
     source('bigquery', 'datarepo-dev-02c59b72', 'hca_dev_99101928d9b14aafb759e97958ac7403__20210830_20210903'),
     source('bigquery', 'datarepo-dev-a6312a94', 'hca_dev_992aad5e7fab46d9a47ddf715e8cfd24__20210830_20210903'),
@@ -168,6 +169,7 @@ dcp2_sources = union({}, 105, delta([
     source('bigquery', 'datarepo-dev-67240cf2', 'hca_dev_f86f1ab41fbb4510ae353ffd752d4dfc__20210901_20210903'),
     source('bigquery', 'datarepo-dev-e8e0a59a', 'hca_dev_f8aa201c4ff145a4890e840d63459ca2__20210901_20210903'),
     source('bigquery', 'datarepo-dev-96d8e08c', 'hca_dev_faeedcb0e0464be7b1ad80a3eeabb066__20210831_20210903'),
+    # @formatter:on
 ]))
 
 dcp3_sources = union(dcp2_sources, 111, delta([

